functorch/functorch/_src/aot_autograd.py

```diff
@@ -197,11 +197,25 @@ def forward(ctx, *flat_tensor_args):
             # TODO - Remove when https://github.com/pytorch/functorch/pull/794 is fixed.
             old_jit_autocast_flag = torch._C._jit_set_autocast_mode(False)
             if compiled_fw is None:
-                flat_tensor_args = pytree.tree_map(
-                    lambda x: x.detach().requires_grad_(x.requires_grad)
-                    if isinstance(x, Tensor) else x, flat_tensor_args
-                )
-                fake_mode = FakeTensorMode.push() if config.use_fake_tensor else nullcontext()
+                if config.use_fake_tensor:
+                    fake_mode = FakeTensorMode.push()
+                else:
+                    fake_mode = nullcontext()
+                    # The detach().requires_grad_() pattern can cause some subtle bugs.
+                    # These will be fixed once FakeTensor is always-on for AOTAutograd.
+                    #
+                    # For models that might resize their inputs, the input tensors
+                    # must have allow_tensor_metadata_change() set to true.
+                    # detach() returns a view tensor, but with that field set to false.
```

Comment:
why don't you just clone lol

Reply:
yeah, I just figured that might lead to other errors (OOM'ing during tracing), and we're hopefully going to kill this soon once FakeTensor mode is always on anyway